discovery-provider/src/tasks/users.py

```diff
@@ -188,27 +190,51 @@ def parse_user_event(
     # New updated_at timestamp
     user_record.updated_at = datetime.utcfromtimestamp(block_timestamp)
 
-    # If creator, look up metadata multihash in IPFS and override with metadata fields
-    metadata_overrides = get_metadata_overrides_from_ipfs(
-        session, update_task, user_record
-    )
+    # If the multihash is updated, fetch the metadata (if not fetched) and update the associated wallets column
+    if event_type == user_event_types_lookup["update_multihash"]:
+         # If creator, look up metadata multihash in IPFS and override with metadata fields
+        ipfs_metadata = get_ipfs_metadata(
+            update_task, user_record
+        )
+
+        if ipfs_metadata:
+            # ipfs_metadata properties are defined in get_ipfs_metadata
+            if "profile_picture" in ipfs_metadata and \
+                ipfs_metadata["profile_picture"]:
+                user_record.profile_picture = ipfs_metadata["profile_picture"]
+
+            if "profile_picture_sizes" in ipfs_metadata and \
+                ipfs_metadata["profile_picture_sizes"]:
+                user_record.profile_picture = ipfs_metadata["profile_picture_sizes"]
+
+            if "cover_photo" in ipfs_metadata and \
+                ipfs_metadata["cover_photo"]:
+                user_record.cover_photo = ipfs_metadata["cover_photo"]
+
+            if "cover_photo_sizes" in ipfs_metadata and \
+                ipfs_metadata["cover_photo_sizes"]:
+                user_record.cover_photo = ipfs_metadata["cover_photo_sizes"]
 
-    if metadata_overrides:
-        # metadata_overrides properties are defined in get_metadata_overrides_from_ipfs
-        if metadata_overrides["profile_picture"]:
-            user_record.profile_picture = metadata_overrides["profile_picture"]
-        if metadata_overrides["profile_picture_sizes"]:
-            user_record.profile_picture = metadata_overrides["profile_picture_sizes"]
-        if metadata_overrides["cover_photo"]:
-            user_record.cover_photo = metadata_overrides["cover_photo"]
-        if metadata_overrides["cover_photo_sizes"]:
-            user_record.cover_photo = metadata_overrides["cover_photo_sizes"]
-        if metadata_overrides["bio"]:
-            user_record.bio = metadata_overrides["bio"]
-        if metadata_overrides["name"]:
-            user_record.name = metadata_overrides["name"]
-        if metadata_overrides["location"]:
-            user_record.location = metadata_overrides["location"]
+            if "bio" in ipfs_metadata and \
```

Comment:
what happens here if i am intentionally setting my bio to empty? will that get updated, or will a prior non-null bio take precedence?

Reply:
i guess as a creator too, since i think non-creators still write to chain directly?

Reply:
you know after taking a closer look here I think there's some weird behavior on current master - since [this](https://github.com/AudiusProject/audius-protocol/blob/764ad083f40a045ac16bf2ac697116e20aff5c91/discovery-provider/src/utils/ipfs_lib.py#L27) only checks for None, and not an empty string, you can actually nullify your bio. I guess we should do the same check here?

Reply:
this is quite funny. prod is actually weird about this. (just learned)

since something like bio is stored both on chain and off chain, we would handle this correctly on this branch as well as on prod. i verified this against my local stack just now (am able to set a bio and then set it to ""). this works because we would index the update_bio event, update the discovery UserRecord.bio, index the update_multihash event, and because bio would be falsey, do nothing.

however this breaks in the case where a metadata field is not a chain field. we even see this now with profile_picture_sizes, but we've never encountered this issue because it's impossible to "unset" your profile picture in the UI (hilarious).

Going forward here, this change is relatively safe for both collectibles and wallets because you would always be setting them back to sensible defaults (empty object), but to make this tighter, for non chain fields, I'll do this for collectibles

```
if "collectibles" in ipfs_metadata and ...:
  user_record.has_collectibles = True
else:
  user_record.has_collectibles = False
```

do you think we should for profile_picture_sizes/similar as well?

Reply:
as for wallets, i could also see the argument for explicitly checking the null case and if so clearing wallets from the new db tables

Reply:
let's definitely do the above for collectibles, and I think we should probably clear the wallets as well - if the latest metadata object no longer contains `associated_wallets` we should ensure any wallets for this user are removed

Reply:
what about the profile_picture_sizes? clear that out too? (happy to)

Reply:
idk, i think its kind of fine to leave it since we literally don't allow you to clear out a profile picture? what's your take here?

Reply:
i think we should probs be consistent if we do wallet & collectibles. in the UI we would probably set collectibles to `{}` and wallets to `{}` when you delete, so that would work as is, but if we're going to be "destructive" i'd say maybe do it across the board?

Reply:
🤷

Reply:
let's go ahead and make it consistent per our offline discussion and set both to empty if we do encounter both - ie 'profile_picture' and 'profile_picture_sizes' are falsey, let's invalidate the values for the user_record